### PR TITLE
Added `autoPause` config for servers

### PR DESCRIPTION
**Pull request by zenonet**
`autoPause` pauses the server while nobody is online.
This is useful especially for private servers because it means, that you won't have to worry about getting destroyed by waves while you are offline. You can just continue playing when you rejoin.

It is implemented as a `Config` and is disabled by default.

image

- [x] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [x] I have ensured that my code compiles, if applicable.
- [x] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

**Comment by Anuken** [comment]
Your IDE reformatted a bunch of unrelated code in ServerControl, this must be fixed.

**Comment by zenonet** [comment]
Done

File: server/src/mindustry/server/ServerControl.java [hunk 1]
```diff
@@ -269,6 +270,27 @@ protected void setup(String[] args){
 
             info("Server loaded. Type @ for help.", "'help'");
         });
+
+        Events.on(PlayerJoin.class, e -> {
+            if(state.serverPaused && autoPaused && Config.autoPause.bool()){
+                state.serverPaused = false;
+                autoPaused = false;
+
+                info("Game was automatically resumed because a player joined.");
```
Comment (Anuken) [inline server/src/mindustry/server/ServerControl.java:279]: This is going to be logged a lot, is it really particularly important to write about either of these events? The server already logs when someone joins/leaves.

Comment (itskatt) [reply server/src/mindustry/server/ServerControl.java:279]: Probably this should be included in the join/leave messages, ex. `Player has joined/left the game, resuming/pausing the game`

**Comment by zenonet** [comment]
I remove the logging for `autoPause` now

**Pull request merged**